esp-hal/src/gpio/mod.rs

```diff
@@ -2043,7 +1965,74 @@ pub(crate) mod internal {
     }
 
     impl InputPin for AnyPin {}
-    impl OutputPin for AnyPin {}
+
+    // Need to forward these one by one because not all pins support all functions.
+    impl OutputPin for AnyPin {
```

Comment:
This was a bug in one of the last PRs, allowing using output APIs with input-only pins.